Xamarin.Forms.Core/VisualElement.cs

```diff
@@ -783,9 +784,26 @@ static void FlowDirectionChanged(BindableObject bindable, object oldValue, objec
 			self.NotifyFlowDirectionChanged();
 		}
 
+		static void OnIsEnabledPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			if (!(bindable is VisualElement element))
+			{
+				return;
+			}
+
+			var isEnabled = (bool)newValue;
+
+			VisualStateManager.GoToState(element, isEnabled 
+				? VisualStateManager.CommonStates.Normal 
+				: VisualStateManager.CommonStates.Disabled);
+		}
+
 		static void OnIsFocusedPropertyChanged(BindableObject bindable, object oldvalue, object newvalue)
 		{
-			var element = bindable as VisualElement;
+			if (!(bindable is VisualElement element))
```

Comment:
check for GO.GetIsDefault(). element will only be a VE, or null

Reply:
Are you saying that GetIsDefault will be faster than the pattern matching for this?

Reply:
Using direct cast now.